pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/TransactionMetricsTest.java

```diff
@@ -339,6 +346,64 @@ public void testManagedLedgerMetricsWhenPendingAckNotInit() throws Exception{
         checkManagedLedgerMetrics(subName2, 32, metric);
     }
 
+    @Test
+    public void testDuplicateMetricTypeDefinitions() throws Exception{
+        pulsarClient = PulsarClient.builder().serviceUrl(lookupUrl.toString()).enableTransaction(true).build();
+        Producer<byte[]> p1 = pulsarClient
+                .newProducer()
+                .topic("persistent://my-property/use/my-ns/my-topic1")
+                .sendTimeout(0, TimeUnit.SECONDS)
+                .create();
+        Transaction transaction = pulsarClient
+                .newTransaction()
+                .withTransactionTimeout(5, TimeUnit.SECONDS)
+                .build()
+                .get();
+        for (int i = 0; i < 10; i++) {
+            String message = "my-message-" + i;
+            p1.newMessage(transaction)
+                    .value(message.getBytes())
+                    .send();
+        }
+        ByteArrayOutputStream statsOut = new ByteArrayOutputStream();
+        PrometheusMetricsGenerator.generate(pulsar, false, false, false, statsOut);
+        String metricsStr = statsOut.toString();
+
+        Map<String, String> typeDefs = new HashMap<>();
+        Map<String, String> metricNames = new HashMap<>();
+        Pattern typePattern = Pattern.compile("^#\\s+TYPE\\s+(\\w+)\\s+(\\w+)");
+
+        Splitter.on("\n").split(metricsStr).forEach(line -> {
```

Comment:
this test are correct? seem not change the code, the test always run success

Reply:
The TC was 1, so this problem arises. I have fixed it